src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -387,20 +389,36 @@ public UUID getFirstSourceDatasetIdFromSnapshotId(
   }
 
   public SnapshotPreviewModel retrievePreview(
-      UUID snapshotId, String tableName, int limit, int offset) {
+      UUID snapshotId,
+      String tableName,
+      int limit,
+      int offset,
+      String sort,
+      SqlSortDirection direction) {
     Snapshot snapshot = retrieve(snapshotId);
 
-    snapshot.getTables().stream()
-        .filter(t -> t.getName().equals(tableName))
-        .findFirst()
-        .orElseThrow(
-            () ->
-                new SnapshotPreviewException(
-                    "No snapshot table exists with the name: " + tableName));
+    SnapshotTable table =
+        snapshot
+            .getTableByName(tableName)
+            .orElseThrow(
+                () ->
+                    new SnapshotPreviewException(
+                        "No snapshot table exists with the name: " + tableName));
+
+    SqlSortDirection sortDirection = Optional.ofNullable(direction).orElse(SqlSortDirection.ASC);
+    if (!StringUtils.equals(sort, PDAO_ROW_ID_COLUMN)) {
+      table
+          .getColumnByName(sort)
+          .orElseThrow(
+              () ->
+                  new SnapshotPreviewException(
+                      "No snapshot table column exists with the name: " + sort));
+    }
 
     try {
       List<Map<String, Object>> values =
-          bigQuerySnapshotPdao.getSnapshotTable(snapshot, tableName, limit, offset);
+          bigQuerySnapshotPdao.getSnapshotTable(
```

Comment:
It looks like this API is only used by this method. If so, you could pass `SnapshotTable` instead of `tableName`. It would be nice to pass in `Column` for `sort` too but I guess the PDAO row id column is a special case and isn't in the schema...

Reply:
I was thinking it's more type safe since it's a more specific type than `String`, so it's harder to accidentally pass the wrong argument for a parameter. But there's not much benefit here, really.